### DISCUSSION TITLE
Trivy license scan results

### DISCUSSION THREAD

**Discussion by RonShvarz** [community/community#61039] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hello.
I've been scanning using trivy in the scope of git actions, with code scanning enabled.
I don't see any results when i upload the Sarif file, and i've verified results are existing when scanning the same file localy.

For example, i scan an image in my CI in the following manner :

>  - name: Run Trivy license scan on image
>         run: trivy image 'docker.io/myimage:v1.2.3'  --scanners license --license-full --severity 'HIGH,CRITICAL' --format sarif  --output trivy_license_image.sarif

And i upload it in the following manner:

> - name: Upload Trivy image license results to GitHub Security tab
        uses: github/codeql-action/upload-sarif@v2
        with:
          sarif_file: 'trivy_license_image.sarif'
          category: 'Trivy-License-Image'

When ran localy, this scan produces the following result :
![image](https://github.com/community/community/assets/34660076/a905cd30-a70d-4d38-ba74-81b77654b196)

My conclusion is that the scan produces results, but the upload doesn't change my results under the security tab of the same repository.

Thanks in advance.

**Comment by sampart** [comment]
Hello @RonShvarz, thanks for using Code Scanning!

I'm one of the engineers on Code Scanning and have been taking a look at your issue.

First off, is this a duplicate of https://github.com/orgs/community/discussions/60982? If so, perhaps you could close that original one? Thanks

To your specific problem, the information that you provided here was really useful, thank you. I was able to find your uploaded SARIF file by using the `category` you provided 😍 Looking at that file, it doesn't actually seem to have any results in. I appreciate that you're seeing output from the tool, but that doesn't seem to be being reflected in the SARIF file that it's creating.

Could you take a look at the SARIF file being produced when you run locally? It's [JSON](https://en.wikipedia.org/wiki/JSON), so reasonably readable (although they can sometimes be long). In particular, I noticed that `"results"` is an empty array (as is `"rules"`, but that matters less).

If you're seeing results data when running locally, could you compare with what the Actions run is producing? You can configure your workflow to make the SARIF file available as a [build artifact](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts), using the [Upload Artifact Action](https://github.com/actions/upload-artifact).

Finally, on https://github.com/orgs/community/discussions/60982, you asked about filtering by category. You're right that this isn't possible - category is primarily used internally for matching repeated runs of a tool and determining how best to compare them. Once you have a populated SARIF file, you may find you can use rule matching to filter between license issues and other Trivy results.

Let me know how you get on. Thanks

**Reply by RonShvarz** [reply]
Hey @sampart , thanks for replying so quickly!
Firsly, i will close the duplicated https://github.com/orgs/community/discussions/60982 issue, since you have addressed the filtering here.
I do seem to get different results when using SARIF file as output format, as to asking for redirecting the output from the command directly to a text-based file.

So right now the main issue is a mismatch between expected results when swithcing formats for uploading to the GHS.

If i will get a rule to filter vulnerability and license, that will be sufficient, so that will solve my latter question.

I will focus on finding out why SARIF provides a different output.
As to what formats are uploadable to GHS, am i limited to SARIF files?
Is there a workaround to upload the command output, or a JSON output?
When running the command with a Json output parameters 
`--format json  --output trivy_license_image.json`
I do indeed get the results i specified in the latter screenshot.

Thanks!

**Reply by sampart** [reply]
Thanks Ron. You're right that GHAS requires SARIF files. I'm asking around to see if any of my colleagues know of a converter that could help you here.

**Reply by sampart** [reply]
Hello again,

I'm afraid we're not aware of any converters for the JSON output. However, I notice that your workflow is invoking Trivy directly. Our [recommended starter workflow for Trivy](https://github.com/actions/starter-workflows/blob/main/code-scanning/trivy.yml) uses https://github.com/aquasecurity/trivy-action so you might like to try that (if it supports license-checking mode).

Otherwise, this sounds like a question for Trivy support. Hope you manage to get to the bottom of this!

**Reply by RonShvarz** [reply]
I've been using the trivy-action for vulnerability scans, but i could not find it to work with licenses, even via the args property in the action.
I've opened an issue for Trivy support which ill attach below, and hopefully add the resolution here in due time.
Thanks Sam.

https://github.com/aquasecurity/trivy/discussions/4836

**Comment by RonShvarz** [comment, marked as answer]
https://aquasecurity.github.io/trivy/v0.43/docs/configuration/reporting/#sarif

SARIF format does not support license scanning at the moment.
